controllers/common/fx.go

```diff
@@ -145,3 +153,54 @@ func Bootstrap(params Params) error {
 
 	return nil
 }
+
+// PickChildCRDPredicate allows events to trigger the Reconcile of Chaos CRD,
+// for example:
+// Reconcile of IOChaos could be triggered by changes on PodIOChaos.
+// For now, we have PodHttpChaos/PodIOChaos/PodNetworkChaos which require to follow this pattern.
+type PickChildCRDPredicate struct {
+	predicate.Funcs
+}
+
+// Update implements UpdateEvent filter for child CRD.
+func (PickChildCRDPredicate) Update(e event.UpdateEvent) bool {
+	switch e.ObjectNew.(type) {
+	case *v1alpha1.PodHttpChaos, *v1alpha1.PodIOChaos, *v1alpha1.PodNetworkChaos:
+		return true
+	}
+	return false
+}
+
+// StatusRecordEventsChangePredicate skip the update event,
+// when we Only update object.status.experiment.records[].events
+type StatusRecordEventsChangePredicate struct {
+	predicate.Funcs
+}
+
+// Update implements UpdateEvent filter for update deletionTimestamp.
+func (StatusRecordEventsChangePredicate) Update(e event.UpdateEvent) bool {
+	objNew, ok := e.ObjectNew.DeepCopyObject().(v1alpha1.StatefulObject)
+	if !ok {
+		return true
+	}
+	objOld, ok := e.ObjectOld.DeepCopyObject().(v1alpha1.StatefulObject)
+	if !ok {
+		return true
+	}
```

Comment:
I prefer to return `false` here.

Because we would use `Or()` for all the predicates, so each predicate would implement the "restricted" way. For this one, it should only allow:
- the event about "Update" on certain Chaos Resource, `v1alpha1.StatefulObject` 
- there are only changes on `status.Experiment.Records[].Events`